newbank/server/NewBank.java

```diff
@@ -1,9 +1,11 @@
 package newbank.server;
 
 import java.util.HashMap;
+import java.util.Scanner;
 
 public class NewBank {
-	
+
+	Scanner reader = new Scanner(System.in);
```

Comment:
This scanner will listen for input and output on the server instead of the client and so unfortunately it is not possible to interact with the user in this manner. 

NewBankClientHandler has properties (out and in) that allow for this interaction. The only thing that gets printed to the client from this currently is "Account has been created", which is returned by addNewAccount and then printed to the client. 

I think for now the first iteration of this feature it would be acceptable to simple follow the protocol and allow the user to enter NEWACCOUNT <name>, where <name> is the name of the account and then return either SUCCESS or FAIL

Reply:
I'm planning to allow it just for the 3 types of accounts listed in the code (Main, savings, and checking), potentially with us following up in the new week on whether other types can be allowed. As for listing the options, perhaps this could also be a follow up to the customer next week? Seems like it may get kind of clunky as we add in the new options, but would be helpful to clarify before adding it.

Does that seem okay to you? Essentially, update the name and set it up for the accounts we know can exist.

Reply:
Yes I agree I think listing options would be better to be done on a later ticket after discussion, I think for now then do you think it would be worth updating the protocol.txt file with the new command so a user doesn't need to go through the code find out how to create an account.

I do think it would make more sense to have all the interaction with the user on the client instead of the server as this saves the user having to flick between the client and the server.

Reply:
Have updated the code to allow for these accounts (failing if already exists, since more than 1 of the same would make things like transfer have issues). Also updated the protocol.txt file to mention the possible names currently.

Reply:
Apologies - I think I may have been going through an old commit so my comments may have been outdated, will go through the newer commits. 

Not sure how everyone else feels about git history so this is more of an opinion / one for the team but instead of having a commit for each amendment would it make more sense to have one commit per feature / change. 

So instead of creating a new commit with the same name the new changes can be amended to the latest commit with the command:

`git commit --amend --no-edit`

Amending the commit has to be done when you have staged changes you want to commit. This can also be achieved retrospectively by squashing the commits, this can be done with an interactive rebase. For example you could use:

`git rebase -i develop`

I would suggest a few of the commits could be squashed into one by selecting picking the most recent and selecting 'fixup' for the others with the same name.

Definitely not required for approval but potentially useful to bear in mind !